### ISSUE TITLE
Notable differences to original Ender Quarry - Quarry ignores chest loot / mines dirt 

### ISSUE THREAD

**Issue by Cakx** [avlad171/Ender-Quarry-Mod#3]
I found 2 notable differences to the original Ender Quarry:

1) Your ported version does not aquire loot from broken chests/containers. Instead only the chests themselves are mined and the loot drops somewhere into the world at the chests former location.

	Original Behavior
		• The content / loot of mined chests is is collected by the Ender Quarry.

2) Your ported version also mines dirt blocks. This also means that you can mine a completed quarry dirt field over and over again when you reset the quarry. A player can use this to create infinite Dirt. 

	Original Behavior:
		• Dirt (and grass) blocks are ignored unless you use the World Hole Upgrade.

-------

Using your Version 1.1.0
Tested on Forge: 14.23.5.2860

Is there any chance you could still fix these two things in your port? Especially the first one has always been a very important feature of the Ender Quarry.

-------

Thank you very much for this Port, have a nice day.
- Cakx

**Comment by avlad171** [comment]
Hello, thanks for the report, I will look into the code but don't have time to do so anytime soon. I'll try to fix the chest loot issue but regarding dirt I don't know what to say, I kind of wanted it to mine dirt without creating holes in the map. Dirt is very cheap and probably renewable in most modpacks too so this wasn't an issue for my SMP. Plus all the duplication glitches and EMC exploits in modded already give players infinite dirt. I'll think about it.

**Comment by Cakx** [comment]
I agree, you are right about the dirt. Having a quarry mining dirt actually sounds pretty useful for terraforming projects. Since a player always pays with energy to fuel the quarry it's also kinda fair.

The chest looting is the only thing that I'd really love to see back on 1.12.2 since I remember how useful this was on a funny playthrough I had on 1.7.10.

**Comment by avlad171** [comment]
After finding some free time I managed to fix the issues. Getting minecraft to compile was harder than writing the actual code. Now for the techinal details, RWTema used in the original ExU a hook for the "onEntityJoinWorld" event which was triggered when items spawned after breaking their storage. To activate it he used a dictionary where all (?) storage blocks were registered. This approach is very nice but in order to find the said storage blocks reflective java code was used and I don't want do deal with that, so I used the ItemHandler capability introduced in newer versions.

In theory this is very easy to use, in practice there can be a few issues:

- If a mod adds a storage block and it does not register it with this capability the items will drop down.
- If a mod implements a more complex storage system where the blocks are just "portals" to a common storage the quarry will drain the entire storage system. Think of an AE system although ME interfaces use a custom capability for that.

I haven't tested because I'm on a tight schedule and getting mods to run in the Eclipse environment is a pain so any feedback will be appreciated.
